modules/serverless-kafka-sink-security-config.adoc

```diff
@@ -0,0 +1,101 @@
+// Module is included in the following assemblies:
+//
+// * serverless/admin_guide/serverless-kafka-admin.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-kafka-sink-security-config_{context}"]
+= Configuring security for Kafka sinks
+
+_Transport Layer Security_ (TLS) is used by Apache Kafka clients and servers to encrypt traffic between Knative and Kafka, as well as for authentication. TLS is the only supported method of traffic encryption for Knative Kafka.
+
+_Simple Authentication and Security Layer_ (SASL) is used by Apache Kafka for authentication. If you use SASL authentication on your cluster, users must provide credentials to Knative for communicating with the Kafka cluster, otherwise events cannot be produced or consumed.
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resources (CRs) are installed on your {product-title} cluster.
+* Kafka sink is enabled in the `KnativeKafka` CR.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+* You have a Kafka cluster CA certificate stored as a `.pem` file.
+* You have a Kafka cluster client certificate and a key stored as `.pem` files.
+* You have installed the OpenShift (`oc`) CLI.
+
+.Procedure
+
+. Create the certificate files as a secret in the same namespace as your `KafkaSink` object:
++
+[IMPORTANT]
+====
+Certificates and keys must be in PEM format.
+====
+
+** For authentication using SASL without encryption:
++
+[source,terminal]
+----
+$ oc create secret -n <namespace> generic <secret_name> \
+  --from-literal=protocol=SASL_PLAINTEXT \
+  --from-literal=sasl.mechanism=<sasl_mechanism> \
+  --from-literal=user=<username> \
+  --from-literal=password=<password>
+----
+
+** For authentication using SASL and encryption using TLS:
++
+[source,terminal]
+----
+$ oc create secret -n <namespace> generic <secret_name> \
+  --from-literal=protocol=SASL_SSL \
+  --from-literal=sasl.mechanism=<sasl_mechanism> \
+  --from-file=ca.crt=<my_caroot.pem_file_path> \ <1>
+  --from-literal=user=<username> \
+  --from-literal=password=<password>
+----
+<1> The `ca.crt` can be omitted to use the system's root CA set. For example, if you are using a public cloud managed Kafka service such as Red Hat OpenShift Streams for Apache Kafka.
+
+** For encryption using TLS without client authentication:
++
+[source,terminal]
+----
+$ oc create secret -n <namespace> generic <secret_name> \
+  --from-literal=protocol=SSL \
+  --from-file=ca.crt=<my_caroot.pem_file_path> \ <1>
+  --from-literal=user.skip=true
+----
+<1> The `ca.crt` can be omitted to use the system's root CA set. For example, if you are using a public cloud managed Kafka service such as Red Hat OpenShift Streams for Apache Kafka.
```

Comment:
so, this bit doesn't work, the docs is correct though, the bug is in the product. I have opened https://issues.redhat.com/browse/SRVKE-1328

should we remove it from docs, or just leave it there and perhaps release-note the issue?

Reply:
Does it work for upstream or should we also remove it there or open a bug at the source?

Reply:
There's an open issue in upstream https://github.com/knative-sandbox/eventing-kafka-broker/issues/2708